### PR TITLE
Optimize seek leader store logic

### PR DESCRIPTION
**Pull request by birdstorm**
remove redundant seek leader logic that may cause retry.

### CODE REVIEW

File: src/main/java/org/tikv/common/region/AbstractRegionStoreClient.java [hunk 1]
```diff
@@ -240,6 +212,12 @@ private Boolean seekLeaderStore(SlowLog slowLog) {
           updateClientStub();
           return true;
         }
```
Comment (zz-jason) [inline src/main/java/org/tikv/common/region/AbstractRegionStoreClient.java:214]: should we add a log here to indicate that the leader is unreachable?

**Review by marsishandsome — APPROVED** [review]
LGTM

**Pull request merged**